de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/negotiation/ProjectNegotiationCollector.java

```diff
@@ -0,0 +1,96 @@
+package de.fu_berlin.inf.dpp.negotiation;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.apache.log4j.Logger;
+
+import de.fu_berlin.inf.dpp.filesystem.IProject;
+import de.fu_berlin.inf.dpp.filesystem.IResource;
+
+/**
+ * This class collects resources that should be added to a session to add them
+ * at once in preparation of a new single project negotiation. The reason is to
+ * prevent multiple concurrent running project negotiations per user.
+ * 
+ * This class allows to hold the reference to a thread that should handle this
+ * new project negotiation.
+ */
+public class ProjectNegotiationCollector {
```

Comment:
Why is this class holding the worker thread?

It collects mappings just fine, the worker thread stuff seems tagged on. Both parts could be separated easily and I do not think the "Collector" is responsible for the worker thread, but rather the SessionManager, which even creates the worker.

I get that this needs to be synchronized with the other methods, but responsibilities are a bit unclear and it would be nice, if this could be cleaned up a bit, as this stuff belongs into the SessionManager.

Reply:
My thought was to group logic to enable calls like `nextProjectNegotiation.isWorkerAlive()`, but rethinking it now... you're right and I moved it.